engine/src/main/java/org/terasology/engine/rendering/opengl/ScreenGrabber.java

```diff
@@ -101,15 +101,13 @@ public void saveScreenshot() {
         int width = sceneFinalFbo.width();
         int height = sceneFinalFbo.height();
 
-        Runnable task;
         if (savingGamePreview) {
-            task = () -> saveGamePreviewTask(buffer, width, height);
+            GameScheduler.boundedElastic().schedule(() -> saveScreenshotTask(buffer, width, height));
```

Comment:
It looks like you accidentally changed this to `saveScreenshotTask`, which is why game previews aren't saving.
```suggestion
            GameScheduler.boundedElastic().schedule(() -> saveGamePreviewTask(buffer, width, height));
```